SDK/AppCenter/Microsoft.AppCenter.Shared/AppCenter.cs

```diff
@@ -11,50 +13,71 @@ namespace Microsoft.AppCenter
     /// </summary>
     public partial class AppCenter
     {
-        // Gets the first instance of an app secret corresponding to the given platform name, or returns the string 
+        const char SecretDelimiter = ';';
+        const char PlatformKeyValueDelimiter = '=';
+        const string TargetPostfix = "Target";
+        const string SecretPostfix = "appSecret";
+
+        // Gets the first instance of an app sceret and/or target token corresponding to the given platform name, or returns the string 
         // as-is if no identifier can be found. Logs a message if no identifiers can be found.
-        internal static string GetSecretForPlatform(string secrets, string platformIdentifier)
+        internal static string GetSecretAndTargetForPlatform(string secrets, string platformIdentifier)
         {
+            var platformTargetIdentifier = platformIdentifier + TargetPostfix;
             if (string.IsNullOrEmpty(secrets))
             {
                 throw new AppCenterException("App secrets string is null or empty");
             }
 
             // If there are no equals signs, then there are no named identifiers, but log a message in case the developer made 
             // a typing error.
-            if (!secrets.Contains("="))
+            if (!secrets.Contains(PlatformKeyValueDelimiter.ToString()))
             {
                 AppCenterLog.Debug(AppCenterLog.LogTag, "No named identifier found in appSecret; using as-is");
                 return secrets;
             }
 
-            var parseErrorMessage = $"Error parsing key for '{platformIdentifier}'";
+            var secretsDictionary = secrets.Split(SecretDelimiter)
+                                           .Select(value => value.Split(PlatformKeyValueDelimiter))
+                                           .GroupBy(p => p[0].Trim() ?? "", StringComparer.OrdinalIgnoreCase)
+                                           .ToDictionary(pair => pair.Key.Trim(), pair => pair.Last().Last().Trim(), StringComparer.OrdinalIgnoreCase);
 
-            var platformIndicator = platformIdentifier + "=";
-            var secretIdx = secrets.IndexOf(platformIndicator, StringComparison.Ordinal);
-            if (secretIdx == -1)
+            var parseErrorMessage = $"Error parsing key for '{platformIdentifier}'";
+            if (secretsDictionary.ContainsKey(TargetPostfix.ToLower()) || secretsDictionary.ContainsKey(SecretPostfix.ToLower()))
             {
-                throw new AppCenterException(parseErrorMessage);
+                AppCenterLog.Debug(AppCenterLog.LogTag, "Found named identifier in the secret. Returning as-is.");
```

Comment:
Align the message with https://github.com/microsoft/appcenter-sdk-dotnet/pull/1176/files#diff-390008ed19e9532de33a029612f1221fR35.
           
```suggestion
                AppCenterLog.Debug(AppCenterLog.LogTag, "Found named identifier in the secret; using as-is.");
```